incubator/command-dump/src/main/java/io/aklivity/zilla/runtime/command/dump/internal/airline/ZillaDumpCommand.java

```diff
@@ -318,253 +457,322 @@ private boolean handleFrame(
-                encodeIpv6Header(buffer, bindingId, streamId ^ 1L, streamId, TCP_HEADER_SIZE);
-                encodeTcpHeader(buffer, streamId ^ 1L, streamId, sequence, 0L, TcpFlag.PSH);
-
-                writePcapOutput(writer, buffer, 0, TCP_HEADER_LIMIT);
+                writeFrame(FlushFW.TYPE_ID, f.originId(), f.routedId(), f.streamId(), f.timestamp(), f, PSH_ACK);
             }
         }
 
-        private void onReset(
-            ResetFW reset)
+        private void onSignal(
+            SignalFW signal)
         {
-            final long bindingId = reset.routedId();
+            if (allowedBinding.test(signal.routedId()))
+            {
+                writeFrame(SignalFW.TYPE_ID, signal.originId(), signal.routedId(), signal.streamId(), signal.timestamp(), signal,
+                    PSH_ACK);
+            }
+        }
 
-            if (allowedBinding.test(bindingId))
+        private void onChallenge(
+            ChallengeFW challenge)
+        {
+            if (allowedBinding.test(challenge.routedId()))
             {
-                final MutableDirectBuffer buffer = writeBuffer;
-                final long streamId = reset.streamId();
-                final long timestamp = reset.timestamp();
-                final long sequence = reset.sequence();
+                final MutableDirectBuffer buffer = new UnsafeBuffer(ByteBuffer.allocate(challenge.sizeof()));
+                final ChallengeFW f = new ChallengeFW.Builder().wrap(buffer, 0, challenge.sizeof()).set(challenge).build();
+                final ExtensionFW extension = f.extension().get(extensionRO::tryWrap);
+                patchExtension(buffer, extension, ChallengeFW.FIELD_OFFSET_EXTENSION);
 
-                encodePcapHeader(buffer, ETHER_HEADER_SIZE + IPV6_HEADER_SIZE + TCP_HEADER_SIZE, timestamp);
-                encodeEtherHeader(buffer);
-                encodeIpv6Header(buffer, bindingId, streamId, streamId ^ 1L, TCP_HEADER_SIZE);
-                encodeTcpHeader(buffer, streamId, streamId ^ 1L, sequence, 0L, TcpFlag.RST);
+                writeFrame(ChallengeFW.TYPE_ID, f.originId(), f.routedId(), f.streamId(), f.timestamp(), f, PSH_ACK);
+            }
+        }
 
-                writePcapOutput(writer, buffer, 0, TCP_HEADER_LIMIT);
+        private void patchExtension(
+            MutableDirectBuffer buffer,
+            ExtensionFW extension,
+            int offset)
+        {
+            if (extension != null)
+            {
+                int streamTypeId = calculateLabelCrc(extension.typeId());
+                buffer.putInt(offset, streamTypeId);
             }
         }
 
-        private void onWindow(
-            WindowFW window)
+        private int calculateLabelCrc(
+            int labelId)
         {
-            final long bindingId = window.routedId();
+            crc.reset();
+            crc.update(resolveLabelAsBytes(labelId));
+            return (int) crc.getValue();
+        }
```

Comment:
This should be cached, as the same input always gives the same output.

Reply:
OK